### PR TITLE
MM-50377: Add staging models for mobile

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

Contains staging model for mobile event. Structure follows [staging best practices by DBT](https://docs.getdbt.com/guides/best-practices/how-we-structure/2-staging).

The following models are added:
- [base_mm_mobile_prod__tracks.sql](https://github.com/mattermost/mattermost-data-warehouse/compare/MM-50377-event-hierarchy-mobile?expand=1#diff-03ace77837d45a18fa9d2f707b7a87113d6f6691d00e6a3ea60fedfe43df98b5) - merges tables together, keeping columns.
-  [stg_mm_mobile_prod__tracks.sql](https://github.com/mattermost/mattermost-data-warehouse/compare/MM-50377-event-hierarchy-mobile?expand=1#diff-1c528ed377cb38df8e2d33c281d0170cb828ffa214df457959981f8e07064da2) - performs renaming in the columns and it's the model to be used by intermediate or mart models.

### CODE REVIEW

File: transform/mattermost-analytics/models/staging/mm_mobile_prod/stg_mm_mobile_prod__tracks.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    })
+}}
+
+WITH tracks AS (
+    SELECT
+        {{ dbt_utils.star(ref('base_mm_mobile_prod__tracks')) }}
+    FROM
+        {{ ref ('base_mm_mobile_prod__tracks') }}
+)
+SELECT
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/staging/mm_mobile_prod/stg_mm_mobile_prod__tracks.sql:13]: Is it missing `category`?

File: transform/mattermost-analytics/models/staging/mm_mobile_prod/_mm_mobile_prod__models.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+version: 2
+
+models:
+  - name: stg_mm_mobile_prod__tracks
+    description: |
+      Reconstructed `tracks` table using custom properties expected to be in the events.
+
+    columns:
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/staging/mm_mobile_prod/_mm_mobile_prod__models.yml:8]: does it need to contain `server_id` as well?

File: transform/mattermost-analytics/models/staging/mm_mobile_prod/stg_mm_mobile_prod__tracks.sql (hunk 1 above)
Comment (ifoukarakis) [reply transform/mattermost-analytics/models/staging/mm_mobile_prod/stg_mm_mobile_prod__tracks.sql:13]: Nice catch! Adding.

File: transform/mattermost-analytics/models/staging/mm_mobile_prod/_mm_mobile_prod__models.yml (hunk 2 above)
Comment (catalintomai) [reply transform/mattermost-analytics/models/staging/mm_mobile_prod/_mm_mobile_prod__models.yml:8]: Just pointing out that it's missing in this columns list, if we plan to have it.

Comment (ifoukarakis) [reply transform/mattermost-analytics/models/staging/mm_mobile_prod/_mm_mobile_prod__models.yml:8]: Ah, sorry, missed that. Thanks, nice catch, added it!

**Review by sri-byte — APPROVED** [review]

**Review by catalintomai — APPROVED** [review]

**Pull request merged**